keras/src/layers/preprocessing/image_preprocessing/random_zoom.py

```diff
@@ -181,7 +188,115 @@ def transform_bounding_boxes(
         transformation,
         training=True,
     ):
-        raise NotImplementedError
+        if backend_utils.in_tf_graph():
+            self.backend.set_backend("tensorflow")
+
+        def _get_transformed_x_y(x, y, transform):
```

Comment:
Please move the functions `_get_transformed_x_y` and `_get_clipped_bbox` to be methods on the layer class rather than inlined function (for better eager performance).